### PR TITLE
kpi resource test task done

### PR DESCRIPTION
**Pull request by 07Himank**
### Describe your changes :

I worked on Kpi resource test.

# 
### Type of change :

- [x] Bug fix

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@mohityadav766

### CODE REVIEW

File: openmetadata-service/src/test/java/org/openmetadata/service/resources/EntityResourceTest.java [hunk 1]
```diff
@@ -14,55 +14,17 @@
...
-import static org.openmetadata.service.Entity.FIELD_FOLLOWERS;
-import static org.openmetadata.service.Entity.FIELD_OWNER;
-import static org.openmetadata.service.Entity.FIELD_TAGS;
-import static org.openmetadata.service.exception.CatalogExceptionMessage.ENTITY_ALREADY_EXISTS;
-import static org.openmetadata.service.exception.CatalogExceptionMessage.entityIsNotEmpty;
-import static org.openmetadata.service.exception.CatalogExceptionMessage.entityNotFound;
-import static org.openmetadata.service.exception.CatalogExceptionMessage.permissionDenied;
-import static org.openmetadata.service.exception.CatalogExceptionMessage.permissionNotAllowed;
-import static org.openmetadata.service.exception.CatalogExceptionMessage.readOnlyAttribute;
+import static org.openmetadata.service.Entity.*;
```
Comment (mohityadav766) [inline openmetadata-service/src/test/java/org/openmetadata/service/resources/EntityResourceTest.java:20]: avoid wild card imports

File: openmetadata-service/src/test/java/org/openmetadata/service/resources/EntityResourceTest.java [hunk 2]
```diff
@@ -14,55 +14,17 @@
...
+import static org.openmetadata.service.exception.CatalogExceptionMessage.*;
```
Comment (mohityadav766) [inline openmetadata-service/src/test/java/org/openmetadata/service/resources/EntityResourceTest.java:21]: remove wild card imports

File: openmetadata-service/src/test/java/org/openmetadata/service/resources/events/WebhookResourceTest.java [hunk 3]
```diff
@@ -13,13 +13,8 @@
...
 package org.openmetadata.service.resources.events;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertNotNull;
-import static org.junit.jupiter.api.Assertions.assertNull;
-import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.openmetadata.service.util.EntityUtil.fieldAdded;
-import static org.openmetadata.service.util.EntityUtil.fieldDeleted;
-import static org.openmetadata.service.util.EntityUtil.fieldUpdated;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (mohityadav766) [inline openmetadata-service/src/test/java/org/openmetadata/service/resources/events/WebhookResourceTest.java:16]: same

File: openmetadata-service/src/test/java/org/openmetadata/service/resources/events/WebhookResourceTest.java [hunk 4]
```diff
@@ -38,11 +33,7 @@
 import org.openmetadata.schema.api.events.CreateWebhook;
 import org.openmetadata.schema.filter.EventFilter;
 import org.openmetadata.schema.filter.Filters;
-import org.openmetadata.schema.type.ChangeDescription;
-import org.openmetadata.schema.type.ChangeEvent;
-import org.openmetadata.schema.type.EventType;
-import org.openmetadata.schema.type.FailureDetails;
-import org.openmetadata.schema.type.Webhook;
+import org.openmetadata.schema.type.*;
```
Comment (mohityadav766) [inline openmetadata-service/src/test/java/org/openmetadata/service/resources/events/WebhookResourceTest.java:36]: wild card -

File: openmetadata-service/src/test/java/org/openmetadata/service/resources/kpi/KpiResourceTest.java [hunk 5]
```diff
@@ -56,12 +48,12 @@ public KpiResourceTest() {
...
 
-  public void setupKpi(TestInfo test) throws IOException {
+  public void setupKpi() throws IOException {
     DataInsightResourceTest dataInsightResourceTest = new DataInsightResourceTest();
     CreateDataInsightChart chartRequest =
         dataInsightResourceTest
-            .createRequest(test)
-            .withName("TestChart")
+            .createRequest()
+            .withName("TestChart" + UUID.randomUUID())
```
Comment (mohityadav766) [inline openmetadata-service/src/test/java/org/openmetadata/service/resources/kpi/KpiResourceTest.java:56]: can use String.format

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/KpiRepository.java [hunk 6]
```diff
@@ -225,14 +225,15 @@ public KpiUpdater(Kpi original, Kpi updated, Operation operation) {
 
     @Override
     public void entitySpecificUpdate() throws IOException {
-      updateFromRelationships(
+      updateToRelationships(
```
Comment (nahuelverdugo) [inline openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/KpiRepository.java:228]: Why have we done this change from `updateFromRelationships` to `updateToRelationships`? The PR says: `add KPI resource test`.

**Review by nahuelverdugo — CHANGES_REQUESTED** [review]

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/KpiRepository.java (hunk 6 above)
Comment (nahuelverdugo) [reply openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/KpiRepository.java:228]: Reviewed with @mohityadav766

**Pull request merged**